platforms/stm32f405/src/can.c

```
@@ -1,374 +1,64 @@
-                if(status == HAL_OK)
-                {
-                    free(tx_header);
-                    return MSG_SENT;
-                }
-                else
-                {
-                    free(tx_header);
-                    return MSG_FAIL;
-                }
-                }
-            else
-            {
-                free(tx_header);
-                return BUFFER_FULL;
-            }
-            break;
-        // case CAN_LINE_3:
-        //     uint32_t free_spots = HAL_CAN_GetTxMailboxesFreeLevel(can3);
-        //     if(free_spots != 0)
-        //     {
-        //         HAL_StatusTypeDef status = HAL_CAN_AddTxMessage(can3, tx_header, message.data, CAN_TX_MAILBOX2);
-        //         if(status == HAL_OK)
-        //         {
-        //             free(tx_header);
-        //             return SUCCESS;
-        //         }
-        //         else
-        //         {
-        //             free(tx_header);
-        //             return FAIL;
-        //         }
-        //         }
-        //     else
-        //     {
-        //         free(tx_header);
-        //         return FAIL;
-        //     }
-        //     break;
-        default:
-            free_spots = HAL_CAN_GetTxMailboxesFreeLevel(can1);
-            if(free_spots != 0)
-            {
-                HAL_StatusTypeDef status = HAL_CAN_AddTxMessage(can1, tx_header, message.data, CAN_TX_MAILBOX0);
-                if(status == HAL_OK)
-                {
-                    free(tx_header);
-                    return MSG_SENT;
-                }
-                else
-                {
-                    free(tx_header);
-                    return MSG_FAIL;
-                }
-                }
-            else
-            {
-                free(tx_header);
-                return BUFFER_FULL;
-            }
-            break;
+        // Filter Configuration Error
+        Error_Handler();
```

Instead of Error_Handler we probably just want to return the HAL error that we got to the application and let application decide what to do with error